lua/rust-tools/inlay_hints.lua

```diff
@@ -34,11 +35,22 @@ end
 
 -- Enable auto hints and set hints for the current buffer
 function M.enable(self)
-  self.enabled = true
+  self.enable = false
+  M.is_enabled = true
   M.enable_cache_autocmd()
   set_all(self)
 end
 
+-- Toggles inlay hints state globally. Uses disable and enable internally
+function M.toggle(self)
+  if self.is_enabled then
+    M.disabled(self)
```

Comment:
```suggestion
    M.disable(self)
```
Isn't this a typo?

Reply:
Oh wow, I didn't double check the hunk, and committed working, but unclean version. 

Good catch, I updated everything in the PR today, so feel free to report any regressions/problems regarding to new implementation, here